src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -291,48 +317,51 @@ public async Task<SemanticModel> GetSemanticModelAsync(CancellationToken cancell
         /// </summary>
         public Document WithSourceCodeKind(SourceCodeKind kind)
         {
-            return this.Project.Solution.WithDocumentSourceCodeKind(this.Id, kind).GetDocument(this.Id);
+            return this.Project.Solution.WithDocumentSourceCodeKind(this.Id, kind).GetDocument(this.Id)!;
```

Comment:
💭 I'd be interesting to discuss how an immutable API could be created in the future to account for this case without needing the suppression.

Reply:
This is just our fault for not making GetDocument throw if you passed it something wrong. There was always ContainsDocument if you weren't sure, but unfortunately a lot does rely on this today...